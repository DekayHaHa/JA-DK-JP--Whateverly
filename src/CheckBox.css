.genre-container {
  width: 230px;
}

.genre-container label {
  display: flex;
  align-items: center;
  margin: 10px 0 10px 0;
}

.genre-container:hover .faux-box {
  background: #a24457;
}

.genre-container:hover .genre-title {
  color: #a24457;
}

.genre-container:active .faux-box {
  opacity: 0.5;
}

.genre-title {
  margin: 0 50px 0 10px;
  color: #fff;
  font-weight: 700;
  font-size: 20px;
  transition: all .2s ease;
}

.faux-box {
  background: #ce63ff;
  width: 30px;
  height: 30px;
  border-radius: 5px;
  transition: all .2s ease;
  position: relative;
}

input[type=checkbox] {
  display: none;
}

input[type=checkbox]:checked ~ div.faux-box:after {
  position: absolute;
  left: 8px;
  top: 2px;
  content: "";
  width: 8px;
  height: 16px;
  border: solid #fff;
  border-width: 0 5px 5px 0;
  transform: rotate(45deg);
}