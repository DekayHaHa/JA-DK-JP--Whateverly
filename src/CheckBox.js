import React from 'react';
import './CheckBox.css';

export default function Footer (props) {
    return (
        <div className="genre-container">
          <label htmlFor={props.name}>
            <input type="checkbox" className={props.class} id={props.name}/>
            <div className="faux-box"></div>
            <p className="genre-title">{props.name}</p>
          </label>
        </div>
    )
}