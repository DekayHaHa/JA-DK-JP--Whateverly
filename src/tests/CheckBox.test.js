import React from 'react';
import ReactDOM from 'react-dom';
import CheckBox from '../CheckBox';
import { shallow } from 'enzyme';

describe('CheckBox', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <CheckBox />
    );
  });
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });

})