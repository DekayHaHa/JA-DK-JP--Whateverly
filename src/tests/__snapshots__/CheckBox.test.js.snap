// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should match snapshop 1`] = `
<div>
  <input
    type="checkbox"
  />
  <label />
</div>
`;
